modules/datastore/src/Plugin/DatasetInfo/DatastoreInfo.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\datastore\Plugin\DatasetInfo;
+
+use Drupal\common\DatasetInfoPluginBase;
+use Drupal\datastore\DatastoreService;
+use Drupal\datastore\Service\Info\ImportInfo;
+use Drupal\datastore\Service\ResourceLocalizer;
+use Drupal\metastore\ResourceMapper;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Plugin implementation of the dataset_info.
+ *
+ * @DatasetInfoPlugin(
+ *   id = "datastore_info",
+ * )
+ */
+class DatastoreInfo extends DatasetInfoPluginBase {
+
+  /**
+   * Import info service.
+   */
+  protected ImportInfo $importInfo;
+
+  /**
+   * Datastore service.
+   */
+  protected DatastoreService $datastore;
+
+  /**
+   * Resource mapper service.
+   */
+  protected ResourceMapper $resourceMapper;
+
+  /**
+   * Constructs a \Drupal\Component\Plugin\PluginBase object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $pluginId
+   *   The plugin_id for the plugin instance.
+   * @param mixed $pluginDefinition
+   *   The plugin implementation definition.
+   * @param \Drupal\datastore\Service\Info\ImportInfo $importInfo
+   *   Import info datastoer service.
+   * @param \Drupal\datastore\DatastoreService $datastore
+   *   Datastore service.
+   * @param \Drupal\metastore\ResourceMapper $resourceMapper
+   *   Resource mapper service.
+   */
+  public function __construct(
+    array $configuration,
+    $pluginId,
+    $pluginDefinition,
+    ImportInfo $importInfo,
+    DatastoreService $datastore,
+    ResourceMapper $resourceMapper,
```

Comment:
ResourceMapper can be retrieved from the DatastoreService, so you could remove` ResourceMapper $resourceMapper,`